### PR TITLE
Fix #55, release v1.2.0.

### LINKED ISSUE #55: `CriticalSection` should not be `Send` or `Sync`
Allowing the `CriticalSection` to be both `Send` and `Sync` allows multiple threads to gain access to a resource by sharing it between threads. For example with `std` this could be done without any `unsafe` code:
```rs
use std::{cell::Cell, thread};
use critical_section::{CriticalSection, Mutex};

static LIST: Mutex<Cell<Vec<i32>>> = Mutex::new(Cell::new(Vec::new()));

fn push_to_list(cs: CriticalSection, iter: impl IntoIterator<Item = i32>) {
    for item in iter {
        let mut list = LIST.borrow(cs).take();
        list.push(item);
        LIST.borrow(cs).set(list);
        thread::yield_now();
    }
}

fn main() {
    critical_section::with(|cs| thread::scope(|scope| {
        scope.spawn(|| {
            push_to_list(cs, 0..100);
        });
        push_to_list(cs, 100..200);
    }));

    critical_section::with(|cs| {
        println!("{:?}", LIST.borrow(cs).take());
    });
}
```

This causes undefined behaviour and often crashes:
```
csbug(12007,0x16b3cb000) malloc: *** error for object 0x11f606290: pointer being realloc'd was not allocated
csbug(12007,0x16b3cb000) malloc: *** set a breakpoint in malloc_error_break to debug
fish: Job 1, 'cargo run' terminated by signal SIGABRT (Abort)
```

### PR DESCRIPTION
**Pull request by Dirbaio**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -16,6 +16,10 @@ pub use self::mutex::Mutex;
 #[derive(Clone, Copy, Debug)]
 pub struct CriticalSection<'cs> {
     _private: PhantomData<&'cs ()>,
+
+    // Prevent CriticalSection from being Send or Sync
+    // https://github.com/rust-embedded/critical-section/issues/55
+    _not_send_sync: PhantomData<*mut ()>,
```
Comment (rursprung) [inline src/lib.rs:22]: isn't `impl !Send` and `impl !Sync` the canonical way to do this? see https://doc.rust-lang.org/nomicon/send-and-sync.html

Comment (Dirbaio) [reply src/lib.rs:22]: it's unstable, needs `#![feature(negative_impls)]`

**Review by MabezDev — APPROVED** [review]
LGTM, thanks!

**Comment by Dirbaio** [comment]
Published v1.2.0, yanked all previous versions.

**Pull request merged**